sdk/keyvault/azure-keyvault-keys/tests/_test_case.py

```diff
@@ -87,79 +62,69 @@ def get_test_parameters(only_hsm=False, only_vault=False, api_versions=None):
-        testcase_func.__name__, parameterized.to_safe_name(api_version), parameterized.to_safe_name(suffix)
-    )
-
-
 def is_public_cloud():
     return (".microsoftonline.com" in os.getenv('AZURE_AUTHORITY_HOST', ''))
 
     
-class KeysTestCase(AzureTestCase):
-    def setUp(self, *args, **kwargs):
+class KeysClientPreparer(AzureRecordedTestCase):
+    def __init__(self, *args, **kwargs):
         vault_playback_url = "https://vaultname.vault.azure.net"
-        hsm_playback_url = "https://managedhsmname.managedhsm.azure.net"
+        hsm_playback_url = "https://managedhsmvaultname.vault.azure.net"
+        self.is_logging_enabled = kwargs.pop("logging_enable", True)
 
         if self.is_live:
             self.vault_url = os.environ["AZURE_KEYVAULT_URL"]
-            self._scrub_url(real_url=self.vault_url, playback_url=vault_playback_url)
-
-            self.managed_hsm_url = os.environ.get("AZURE_MANAGEDHSM_URL")
+            self.vault_url = self.vault_url.rstrip("/")
+            self.managed_hsm_url = os.environ.get("AZURE_MANAGEDHSM_URL", None)
             if self.managed_hsm_url:
-                self._scrub_url(real_url=self.managed_hsm_url, playback_url=hsm_playback_url)
+                self.managed_hsm_url = self.managed_hsm_url.rstrip("/")
         else:
             self.vault_url = vault_playback_url
             self.managed_hsm_url = hsm_playback_url
 
         self._set_mgmt_settings_real_values()
-        super(KeysTestCase, self).setUp(*args, **kwargs)
 
-    def tearDown(self):
-        HttpChallengeCache.clear()
-        assert len(HttpChallengeCache._cache) == 0
-        super(KeysTestCase, self).tearDown()
+    def __call__(self, fn):
+        def _preparer(test_class, api_version, is_hsm, **kwargs):
 
-    def create_key_client(self, vault_uri, **kwargs):
-        if kwargs.pop("is_async", False):
-            from azure.keyvault.keys.aio import KeyClient
+            self._skip_if_not_configured(api_version, is_hsm)
+            if not self.is_logging_enabled:
+                kwargs.update({"logging_enable": False})
+            endpoint_url = self.managed_hsm_url if is_hsm else self.vault_url
+            client = self.create_key_client(endpoint_url, api_version=api_version, **kwargs)
+
+            with client:
+                fn(test_class, client, is_hsm=is_hsm, managed_hsm_url = self.managed_hsm_url, vault_url = self.vault_url)
+        return _preparer
+        
 
-            credential = self.get_credential(KeyClient, is_async=True)
-        else:
-            from azure.keyvault.keys import KeyClient
 
-            credential = self.get_credential(KeyClient)
+    def create_key_client(self, vault_uri, **kwargs):
+       
+        from azure.keyvault.keys import KeyClient
+
+        credential = self.get_credential(KeyClient)
+        
         return self.create_client_from_credential(KeyClient, credential=credential, vault_url=vault_uri, **kwargs)
 
     def create_crypto_client(self, key, **kwargs):
```

Comment:
Same comments from the async version: we can remove `create_crypto_client` and `_get_attestation_uri`